2018/002-update-channels.md

```diff
@@ -0,0 +1,243 @@
+the availability of a new version. This grain is set by the service that runs on each worker.
+
+#### Salt back-end
+
+##### Triggering updates/migrations
+
+We will continue to use `transactional-update` to trigger the updates and migrations.
+Salt will be used to call `transactional-update migrate`
+
+We can reuse the update orchestration with an extra option to call `transactional-update migrate`.
+
+`transactional-update` will report the result back to velum via grains.
+
+##### Updating in batches
+
+Updating in batches reduces the chances that salt might timeout, and relaxes the network connection.
+
+As clusters can have a large number of nodes, it would create a bottleneck if we run the migration
+on all nodes at once, due to network congestion because of the download of the packages, which can
+be quite large.
+
+Therefore we should have a logic in salt that runs the migration on a set (fixed number) of nodes
+at a time, before going to the next batch.
+
+Salt will have to actively wait for the migrations to finish, and then set the specific grain to
+done once it finished.
+
+#### General considerations for Phase 1
+
+##### Out of sync nodes
+
+There needs to be a way to get a single node up-to-date with maintenance updates and
+new migration targets, especially when a single node is out of sync with updates, due to
+having been powered off for a longer time, or if the customer added a new node from
+an old ISO image to the cluster.
+
+##### Systems Registered against RMT/SMT
+
+If the system is registered against SMT or RMT, we can access the information, that there
+is a new version, but that this version is not completely mirrored to SMT/RMT.
+So we need an additional information: a new version is available, but the admin needs to
+mirror this first on SMT/RMT.
+
+### Phase 2 (2nd iteration)
+
+#### User Interface
+
+* explicit trigger of maintenance update installation (snapshot creation and reboot)
+  - before that change, a user can only trigger the reboot, snapshot/installation happens automatically
+* direct display of update content in velum through parsing the release notes
+* display of update history via velum
+
+#### Manual/automatic update strategy
+
+A customer can choose conservative or progressive update strategy:
+
+* Run maintenance updates manually
+* Run maintenance updates automatically, but switch channels manually
+* Run maintenance updates and switch to update channels automatically (always bleading edge)
+
+#### General considerations for Phase 2
+
+##### Being outdated before a migration
+
+We need to check if the installed packages are good enough to do the migration or not.
+If it's good enough, we can either continue with the migration directly or apply normal
+updates and prolong the migration.
+The user should not be enforced to run all maintenance updates first, because the migration
+would contain/overwrite the patches anyways.
```

Comment:
I'm not sure we can state that. I feel like it highly depends on each minor release:

 * if we respin a new media the user won't have to apply all the maintenance updates before migrating
 * if we do not respin a new media the user must apply all the maintenance updates before migrating